techmatters/flex-plugins
# CHI-1993 Modifications to Yellow banner

Primary reviewer:

## Description
- This PR introduces significant updates to the yellow banner's text display logic and structure with shorter content and adding icons instead of the channel type string for prefixing the identifier. The changes apply in both cases whether the helpline has the client_profiles service flag turned on. (see picture below)
- This work temporarily hardcodes the number of clients (Z) until the backend supports multiple clients for one identifier.
- 'View Records' is replaced by adding links to contacts, cases, and clients respectively
- In the `0 contacts/cases` scenario, the respective string is hidden, and singular and plural forms are supported.
- Translation strings are updated across most languages. Some need attention from the product team.
- This work affects all helplines

### Checklist

### Related Issues
Fixes # https://tech-matters.atlassian.net/browse/CHI-1993

### Verification steps
![Screenshot 2024-02-01 at 11 48 25 AM](https://github.com/techmatters/flex-plugins/assets/102122005/70eb7319-aa06-4460-8f32-9cc3e849cf6e)
^above is modica. icon is the same as sms. Ignore the lack of comma. this has been corrected in code
![Screenshot 2024-02-01 at 12 02 11 PM](https://github.com/techmatters/flex-plugins/assets/102122005/7e70f89a-160c-4002-9ca3-ed3b5501ee23)

- Start a contact from any channel - better to test as many as possible
- Check the yellow banner for correct content and behavior in how the contacts and cases are displayed and interact (opens the correct modal)
- Check both cases of client_profiles flag - this logic can be hard-coded by switching logic [here](https://github.com/techmatters/flex-plugins/pull/2045/files#diff-eadb9912a46353712477e051ccb5e8f3f99b4ec70b43725afdac515361e7bdaf)
